clients/roscpp/include/ros/connection.h

```diff
@@ -210,6 +210,8 @@ class ROSCPP_DECL Connection : public boost::enable_shared_from_this<Connection>
   bool is_server_;
   /// Have we dropped?
   bool dropped_;
+  /// Synchronizes dropped_
+  boost::recursive_mutex dropped_mutex_;
```

Comment:
```
  /// Synchronizes drop() calls
  boost::recursive_mutex drop_mutex_;
```

above should be also adjusted as following? comment is not exactly correct though.

for example,

```
  /// Synchronizes drop_signal_
  boost::recursive_mutex drop_signal_mutex_;
```

Reply:
Thanks for your suggestion

Reply:
This breaks ABI which we are trying to avoid in an already released distro.

Reply:
sorry i missed that aspect. you are right.

Reply:
> This breaks ABI

OK, I'll not add a new member for `Connection`. If I can find a better solution, I'll create a new PR.
So close it now.